jwst/datamodels/model_base.py

```diff
@@ -305,7 +308,7 @@ def _drop_array(d):
         _drop_array(self._instance)
 
     def close(self):
-        if not self._iscopy and self._asdf is not None:
+        if not self._iscopy and getattr(self, '_asdf', None) is not None:
```

Comment:
`self._asdf` should always exist, as it contains the `AsdfFile` instance.  When does it not exist?

Reply:
Hmmm. there was a test which required this, but reverting is not finding a failing test. It may be this change should actually be part of the FITS-hash PR. I will revert this and we'll wait and see if it crops up again.